nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,145 @@
+//import
+const express = require("express");
+const router = express.Router();
+const meals = require("./../data/meals.json");
+
+router.get("/", async (req, res) => {
+  let data = meals;
+
+  if ("date" in req.query ) {
+    try {
+      data = data.filter((meal) =>
+    new Date(meal.createdAt).getTime() > new Date(req.query.date).getTime());
+    if (data.length > 0) {
+      res.json(data)
+    }else{
+      res.status(404);
+      res.json("there is no such meal date,Pls check");
+      
+    }
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  if ("maxPrice" in req.query) {
+    data = meals.filter(meal => Number(meal.price) <= Number(req.query.maxPrice));
+    if (data.length > 0) {
+            res.json(data);
+          } else {
+            res.status(404);
+            res.json("there is no such meal id,Pls check");
+            
+          }
+  }
+
+  if ("title" in req.query) {
+    data = meals.filter(meal => (meal.title).toLowerCase().includes((req.query.title).toLowerCase()));
+    if (data.length > 0) {
+            res.json(data);
+          } else {
+            res.status(404).json("Sorry,No such meals in the menu");
+          }
+  }
+  
+  if ("limit" in req.query) {
+    const mealsNew = meals.map(meal => meal);
+    const limitNo = Number(req.query.limit);
+    data = mealsNew.slice(0, limitNo);
+    if (limitNo <= mealsNew.length && limitNo > 0) {
+            res.json(data);
+          } else {
+            res.status(404).json(`Sorry,there is only ${mealsNew.length} meals left.`);
+          }
+  }
```

Comment:
here I do not want to change the original meals.  Because I will use slice(). It will change the original one.  That is why  I make the array to the new one and change from the new array.